sdk/storage/src/blobs/block_blob_client.cpp

```diff
@@ -75,6 +77,64 @@ namespace Azure { namespace Storage { namespace Blobs {
         options.Context, *m_pipeline, m_blobUrl.ToString(), content, protocolLayerOptions);
   }
 
+  BlobContentInfo BlockBlobClient::UploadFromBuffer(
+      const uint8_t* buffer,
+      std::size_t bufferSize,
+      const UploadBlobOptions& options) const
+  {
+    constexpr int64_t c_defaultBlockSize = 8 * 1024 * 1024;
+    constexpr int64_t c_maximumNumberBlocks = 50000;
+    constexpr int64_t c_grainSize = 4 * 1024;
+
+    int64_t chunkSize = c_defaultBlockSize;
+    if (options.ChunkSize.HasValue())
+    {
+      chunkSize = options.ChunkSize.GetValue();
+    }
+    else
+    {
+      int64_t minBlockSize = (bufferSize + c_maximumNumberBlocks - 1) / c_maximumNumberBlocks;
+      chunkSize = std::max(chunkSize, minBlockSize);
+      chunkSize = (chunkSize + c_grainSize - 1) / c_grainSize * c_grainSize;
+    }
+
+    std::vector<std::pair<BlockType, std::string>> blockIds;
+    auto getBlockId = [](int64_t id) {
+      constexpr std::size_t c_blockIdLength = 64;
+      std::string blockId = std::to_string(id);
+      blockId = std::string(c_blockIdLength - blockId.length(), '0') + blockId;
```

Comment:
Please reference track 1 or other languages for block ID format.

Reply:
I know they are different, but which is better than which?

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/246